netsync/manager.go

```diff
@@ -229,26 +230,40 @@ func (sm *SyncManager) startSync() {
 	}
```

Comment:
This is probably solved somewhere else but I didn't quite get it yet. With this condition, if we ever get a bad peer, we will never leave it. Is that right? I would expect a stall to happen eventually if so.

Reply:
There is a similar test in `handleNewPeerMessage()` that means `startSync` will never be called as long as we think we have a sync peer.

```
	if isSyncCandidate && sm.syncPeer == nil {
		sm.startSync()
	}
```

Reply:
Sorry. I see it can be disconnected internally in `peer.go` at `case <-stallTicker.C:`. Assuming that works in all types of misbehavior, we should not get stuck on a bad syncpeer. If we are still getting stuck, then maybe `case <-stallTicker.C:` code needs to be more robust?

Reply:
Ok yes. Then the stallTicker could be made more robust for that probably. Want me to file an issue saying as much?

Reply:
Sure

Reply:
Ok thanks. Hope [this](https://github.com/gcash/bchd/issues/98) works. Any feedback is appreciated.